app/src/main/java/com/tari/android/wallet/model/TariWalletAddress.kt

```diff
@@ -32,75 +32,47 @@
  */
 package com.tari.android.wallet.model
 
-import android.os.Parcel
 import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
 import java.io.Serializable
 
 /**
  * This wrapper is needed for id parameters in AIDL methods.
  *
  * @author The Tari Development Team
  */
-class TariWalletAddress() : Parcelable, Serializable {
+@Parcelize
+data class TariWalletAddress(val hexString: String = "", val emojiId: String = "") : Parcelable, Serializable {
 
-    var hexString = ""
-    var emojiId = ""
-
-    constructor(hexString: String, emojiId: String) : this() {
-        // crunch fix for not crashing on action related to wallet address
-        if (hexString == zeroHex) {
-            this.hexString = zero66Hex
-            this.emojiId =
-                "\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF57"
-        } else {
-            this.hexString = hexString
-            this.emojiId = emojiId
-        }
-    }
-
-    fun isZeros(): Boolean = hexString == zeroHex || hexString == zero66Hex || hexString.all { it == '0' }
+    fun isZeros(): Boolean = hexString == HEX_ZERO || hexString == HEX_ZERO_66 || hexString.all { it == '0' }
 
     override fun equals(other: Any?): Boolean = (other is TariWalletAddress) && hexString == other.hexString
 
     override fun hashCode(): Int = hexString.hashCode()
 
     override fun toString(): String = "TariWalletAddress(hexString='$hexString', emojiId='$emojiId')"
 
-    // region Parcelable
-
-    constructor(parcel: Parcel) : this() {
-        readFromParcel(parcel)
-    }
-
-    companion object CREATOR : Parcelable.Creator<TariWalletAddress> {
-
-        const val zeroHex = "0000000000000000000000000000000000000000000000000000000000000026"
-        const val zero66Hex = "000000000000000000000000000000000000000000000000000000000000000026"
-
-        override fun createFromParcel(parcel: Parcel): TariWalletAddress {
-            return TariWalletAddress(parcel)
-        }
-
-        override fun newArray(size: Int): Array<TariWalletAddress> {
-            return Array(size) { TariWalletAddress() }
+    companion object {
+        const val HEX_ZERO = "0000000000000000000000000000000000000000000000000000000000000026"
+        const val HEX_ZERO_66 = "000000000000000000000000000000000000000000000000000000000000000026"
```

Comment:
That are strange constants. Last byte can be different depending on the network. So this can generate potential issues in the future.